### PR TITLE
Tests and ci

### PR DESCRIPTION
**Pull request by pavelkharitenko**
# The initial workflow setup for tests, regarding #1 and #2

### **BuildPreCICE.yml:** - makes sure the core package compiles
- installs Julia 1.6
- builds PreCICE.jl

### **RunTests.yml:** - runs unit tests
- installs Julia 1.6
- installs preCICE 2.2.0
- builds PreCICE.jl
- runs Tests of PreCICE.jl

### **TestJuliaVersions.yml** - runs tests on different Julia versions
- Installs various Julia versions (1.4 to nightly releases) and tries everything below on all versions
- installs preCICE
- builds PreCICE.jl
- runs Tests of PreCICE.jl

### CODE REVIEW

**Comment by pavelkharitenko** [comment]
The workflow **TestJuliaVersions.yml** fails on Julia versions we don't support. This workflow should later be triggered only on a crontab scedule or through a manuel start instead of on every push, since new Julia versions don't release that frequent. The manual **workflow_dispatch** trigger works right now only if the test is on the `master` branch.

**Comment by pavelkharitenko** [comment]
@IshaanDesai can you please take a look at this pull request and rate if we should merge this or develop the workflow further?

**Comment by IshaanDesai** [comment]
> The workflow TestJuliaVersions.yml fails on Julia versions we don't support. This workflow should later be triggered only on a crontab scedule or through a manuel start instead of on every push, since new Julia versions don't release that frequent.

If we already know that our bindings do not support certain versions of Julia then lets just test for the versions we intend the bindings to be compatible with. Having a failing workflow is not intended because if we know that certain Julia versions are not compatible then we will just state that in the documentation and not test for those versions anymore. Can you please modify the test accordingly? 

As for setting up the workflow as a cron job, you can get the syntax using a [generator tool](https://crontab-generator.org/) where you can state that for example: *run the job every Sunday at midnight*. Once you have the syntax you can add it to the workflow [in this manner](https://github.com/precice/python-bindings/blob/develop/.github/workflows/build-env.yml#L5-L6).

File: .github/workflows/TestJuliaVersions.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build on different Julia versions
```
Comment (IshaanDesai) [inline .github/workflows/TestJuliaVersions.yml:1]: Clearly describe which Julia versions are being tested on

File: test/runtests.jl [hunk 2]
```diff
@@ -2,6 +2,10 @@ using PreCICE
 using Test
 
 
+libprecicePath = split(readlines(`whereis libprecice`)[], ' ')[2]
+
+PreCICE.setPathToLibprecice(string(libprecicePath))
```
Comment (IshaanDesai) [inline test/runtests.jl:7]: Why is this necessary at the start of the tests?

**Review by IshaanDesai — CHANGES_REQUESTED** [review]
Some minor things, rest looks good :+1:

File: test/runtests.jl (hunk 2 above)
Comment (pavelkharitenko) [reply test/runtests.jl:7]: The path to the libprecice library is set here, to be independant of the preCICE installation location in case one who runs the tests has installed it to a different location.

**Review by IshaanDesai — APPROVED** [review]
